data-prepper-plugins/failures-common/src/main/java/org/opensearch/dataprepper/plugins/dlq/s3/KeyPathGenerator.java

```diff
@@ -0,0 +1,86 @@
+package org.opensearch.dataprepper.plugins.dlq.s3;
```

Comment:
Looks like this is somewhat duplicate of the code in OpenSearchSink. Also Wipro is writing some code that seems to be similar to this. We should have a plan to consolidate this.

Reply:
100%, it is slightly similar but there are differences. I think we should standardize this. There was talk here #2424 as well for providing a way to provide patterns for fetching keys. I am thinking we should you at this holistically at how to we build a framework or tooling that allows any plugin to leverage some of these features.